lib/contractWrapperBase.ts

```diff
@@ -184,9 +184,7 @@ export abstract class ContractWrapperBase implements IContractWrapper {
 
     const currentNetwork = await Utils.getNetworkName();
 
-    const web3 = await Utils.getWeb3();
-
-    const maxGasLimit = await computeMaxGasLimit(web3);
+    const maxGasLimit = await UtilsInternal.computeMaxGasLimit();
 
     if (currentNetwork === "Ganache") {
       return maxGasLimit; // because who cares with ganache and we can't get good estimates from it
```

Comment:
Gas estimation for Ganache should be the same as other networks.
In general, we should strive to execute same arc.js code regardless that networks it is connected to.(as possible as can be done). 
This line should be removed.

Reply:
Ganache returns gas estimates that are not sufficient.  At the time I wrote this code I spent a lot of time on this issue.  I just now gave it another try, but encountered the same result.  I am in the process of moving to truffle 5.0 and web3 1.0 -- maybe something will change then, I can retry when I have things working enough to test.

I think this may be a case requiring network-specific code.

Reply:
Is this a known ganache issue ? If it is please add a link to this issue.

Reply:
https://github.com/trufflesuite/ganache-core/issues/151

Reply:
could you please add a link to that on the code ?